rfcs/006 - Sync changes across databases.md

```diff
@@ -0,0 +1,64 @@
+# 006 - Sync changes across databases RFC
+
+### Proposed
+
+2022-12-15
+
+### Accepted
+
+YYYY-MM-DD
+
+#### Approvers
+
+- David Ellis <david@iasql.com>
+- Yolanda Robla <yolanda@iasql.com>
+- Alejandro Guillen <alejandro@iasql.com>
+- Mohammad Pabandi <mohammad@iasql.com>
+
+### Implementation
+
+- [ ] Implemented: [One or more PRs](https://github.com/iasql/iasql-engine/some-pr-link-here) YYYY-MM-DD
+- [ ] Revoked/Superceded by: [RFC ###](./000 - RFC Template.md) YYYY-MM-DD
+
+## Author(s)
+
+- Luis Fernando De Pombo <luisfer@iasql.com>
+
+## Summary
+
+An IaSQL database connects to a cloud account and automatically backfills it with what is already in the cloud account during the initial onboarding and keeps it up to date over time. However, once you import your staging *and* production cloud accounts to two different databases it is not clear how to make changes to both or keep them in sync. The common workflow is to make changes to your staging deployment and then replicate the changes to your production deployment. One of the appeals of SDK-based IaC tools like Pulumi is that you can apply the blueprint like a hammer with some parametrization. This is also possible in Terraform with HCL, but it is a bit more clunky.
+
+We have leaned deeper into the infrastructure as data workflow by dropping migrations from our deployments and favoring two-way mode with transactions over IaC’s apply. This plays to our strengths and provides a compelling workflow to manage infrastructure. However, we need a good story to import or replicate changes from one database to the other since most people manage staging and production as separate cloud accounts.
+
+One important requirement to keep in mind is being able to exclude certain resources from this special sync. In other words, there might be resources in staging that people don't want to replicate in production and vice versa. This requirement can be sidestepped initially, but whatever design we pick should be able to support this down the line.
+
+## Proposal
+
+TBD
+
+## Alternatives Considered
+
+There are several types of solutions to replicate staging infrastructure changes in production.
+
+### Dump a database and import the dump into another database
+
+An `export` route in the engine implemented using `pgdump` with logic to attempt to make it generic like excluding `aws_account` from the dump and so on. The dump is then intended to be passed into an `import` route for another database. `import` was deprecated for the time being due to the sheer amount of special-casing, which compounded per module, that had to be built within the functionality of export and import.
+
+### "Sync" a database to a different cloud account than the one it is connected to
```

Comment:
The issue I see with this approach is that it might be easy for the sync job to fail due to duplicated FKs. Some resources need a unique name as an identifier and if they do not have a good name convention they will collide. Also, resources like security groups would not be easy to handle.